node/src/components/small_network/counting_format.rs

```diff
@@ -38,15 +53,35 @@ pub(super) struct CountingFormat<F> {
     /// The actual serializer performing the work.
     #[pin]
     inner: F,
```

Comment:
Can you explain why `inner` has to be pinned?

Reply:
Because I want to call things like [`Deserializer::deserialize`](https://docs.rs/tokio-serde/0.8.0/tokio_serde/trait.Deserializer.html#tymethod.deserialize), which take a `self: Pin<&mut Self>, ` :)

Reply:
Ok, I thought it's the other way around – b/c you used `#[pin]` the self is now `Pin<…>`.